---
title: 'Genetic Nurturing Effects: Simulations w/o IGE'
author: "Melisa Chuong"
date: "03/02/2021"
output: 
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# SIMULATION ANALYSES {.tabset}  

Simulations were conducted to explore how results of the pathway models outlined above differed using PGSs with varying levels of predictive ability. 10,000 trio members with respective genetic, phenotypic and PGS variables were simulated. 

## SIMULATING TRIO FAMILIES

Simulations were conducted to explore how results of the pathway models outlined above differed using PGSs with varying levels of predictive ability. 10,000 trio members with respective genetic, phenotypic and PGS variables were simulated. The aim here is to simulate the genetic and phenotypic associations between members of a trio i.e. biological parents and a single offspring. To keep analyses simple and efficient we simulate the trio effects in variances rather than simulating individual genotypes. Transmitted and non-transmitted genetic variances **are not** separated in these simulations.  

I aim to simulate these variables as they would occur in real life where possible, within the constraints of choosing to simulate focusing on variances rather than individual genotypes. Please read the comments within the code for clarity.  

A dataframe with trio phenotypes and respective trio genetics and PRSs is generated. This will then go on to be used in subsequent regression and pathway analyses.  

**What do we simulate**  

Simulated phenotypic variables had a variance of 1. Additive genetic variance was computed as the heritability multiplied by the phenotypic variance. Genetic variance was represented as two separate entities; tagged genetic variance, aiming to capture variance attributable to genotyped variants, and non-tagged genetic variance, aiming to capture variance attributable to non-genotyped variants. The tagged genetic variance was calculated as the additive genetic variance multiplied by a pre-specified value, resulting in the desired tagged genetic variance (this can also be thought of as SNP heritability when using real data). Non-tagged genetic variance would be the remaining genetic variance and was calculated as the difference between the additive genetic variance and tagged genetic variance. All remaining variance was attributed to environmental variance, calculated as the product of the phenotypic variance and the difference between one and the specified heritability value. An additional noise variance was specified aiming to capture inaccuracies/biases in variant beta effect sizes specified in GWAS due to the lack of statistical power, which go onto add noise to PGSs. This noise variance was assigned values starting from zero, with zero representing completely accurate beta effect sizes from GWAS.   

The parental genetic variables were constructed as the sum of tagged and non-tagged genetic counterparts. Tagged genetic variable values were simulated from a normal distribution with a mean of zero, and standard deviation of the square root of the tagged genetic variance (explained above). Similarly, non-tagged genetic variable values were simulated from a normal distribution with a mean of zero, and a standard deviation of the square root of non-tagged genetic variance. 
Similarly, the offspring genetic variables were constructed as the sum of tagged and non-tagged genetic counterparts. Offspring tagged and non-tagged genetic variables were simulated as the average of the summed respective tagged and non-tagged maternal and paternal genetic variables, with the addition of a respective tagged and non-tagged segregation variable aiming to capture variability that occurs from random segregation of genes observed during meiosis (Yanowitz, 2010). The tagged and non-tagged segregation variable values were simulated from a normal distribution with a mean of zero and standard deviation of the square root of half the tagged and non-tagged genetic variance, respectively. 
Separate environmental variables were constructed for each member of the trio. This variables were simulated from a normal distribution with a mean of zero, and a standard deviation of the square root of the environmental variance.
The phenotypic variables for each member of the trio was then constructed as the sum of the respective trio member’s genetic and environmental variable values.
  

Polygenic score variables were derived from the trio member's respective tagged genetic component, as PGSs can only be derived using genotyped variants when using real data. The PGSs also included a noise term aiming to represent noise that can arise from GWAS lacking power, resulting in beta estimates (effect sizes) of each variant being potentially inaccurate/biased. Parental noise variables were simulated from a normal distribution with a mean of zero, and a standard deviation of the square root of the noise variance (specified above). Maternal and paternal PGSs were subsequently calculated as the sum of the respective tagged genetic and noise variables. 
The noise component for the offspring PGSs was calculated as the average of the parental noise variables in order to capture noise correlation seen within families; as it is likely that families share a greater number of genetic variants, thus, sharing a greater number of variants with potentially biased estimates within their PGSs than with non-related individuals. Moreover, offspring PGSs further capture an independent noise component, simulated from a normal distribution with a mean of zero, and a standard deviation of the square root of half of the pre-specified noise variance. Thus, the offspring PGSs were calculated as the sum of the offspring tagged genetic, the average of the parental noise variables and the offspring’s independent noise variables.

Different scenarios, with 15 replications each, were conducted with a range of plausible trait heritabilities (0.3, 0.6, 0.9), proportion of tagged genetic variances (0.2, 0.6, 1) and polygenic score noise (0-1, by 0.1 increments) in order for the findings to have broad applicability to the study of a wide variety of traits/complex trait architectures.   

First I will write a function to create the dataframes which will be named *cohort*. Separate dataframes for each specified scenario will be created with 15 replications.  

```{r, eval=FALSE}

  ##########################################################################
  #1. Coding a function that returns the simulated variables in a dataframe#
  ##########################################################################
  
  set.seed(14) #14 is my lucky number

  sim <- function(h2, tagged, prs_noise){
  
  V_P <- 1 #phenotypic variance is = 1
  V_A <- h2*V_P #the heritability is equal to the additive genetic variance
  V_Tagged <- tagged*V_A #the value of tagged should be SNP heritability i.e. the proportion of the genetic variance that is actually tagged by measured variants
  V_nTagged <- V_A - V_Tagged
  V_E <- (1-h2)*V_P #the environmental variance
  n <- 10000 # simulating 10,000 trios
  
  #Creating a parental genetic variables that captures ALL tagged genetic variance
  maternal_genetic_tagged = rnorm(n, mean=0, sqrt(V_Tagged))
  paternal_genetic_tagged = rnorm(n, mean=0, sqrt(V_Tagged))
  
  #Creating a parental genetic variable that captures ALL non-tagged genetic variance 
  maternal_genetic_ntagged = rnorm(n, mean=0, sqrt(V_nTagged))
  paternal_genetic_ntagged = rnorm(n, mean=0, sqrt(V_nTagged))
  
  #Creating a parental genetic variables that is the sum of the tagged and non-tagged genetic variance
  maternal_genetic = maternal_genetic_tagged + maternal_genetic_ntagged
  paternal_genetic = paternal_genetic_tagged + paternal_genetic_ntagged
  
  #Creating parental environmental variables that capture the environmental variance
  maternal_environment = rnorm(n, mean=0, sd = sqrt(V_E))
  paternal_environment = rnorm(n, mean=0, sd = sqrt(V_E))
  
  #Creating parental phenotypic variables that is made up of both genetic and environmental variance
  maternal_phenotype = maternal_genetic + maternal_environment
  paternal_phenotype = paternal_genetic + paternal_environment
  
  #offspring segregation is generated to capture noise that occurs from genetic segregation
  #separate segregation terms are generated using tagged and non-tagged genetic variance
  offspring_segregation_tagged = rnorm(n, 0, sd=sqrt(V_Tagged/2))
  offspring_segregation_ntagged = rnorm(n, 0, sd=sqrt(V_nTagged/2))
  
  #separate offspring genetic counterparts are created using tagged and non-tagged genetic variance (including segregation variance) from each parent
  offspring_genetic_tagged = (maternal_genetic_tagged + paternal_genetic_tagged)/2 + offspring_segregation_tagged
  offspring_genetic_ntagged = (maternal_genetic_ntagged + paternal_genetic_ntagged)/2 + offspring_segregation_ntagged
  
  #final offspring genetic variable is created by summing the offspring genetic counterparts generated above
  offspring_genetic = offspring_genetic_tagged + offspring_genetic_ntagged
  
  #offspring environmental variable made up of environmental variance
  offspring_environment <- rnorm(n, mean=0, sd = sqrt(V_E))
  
  #offspring phenotypic variable is the sum of the offspring genetic and environmental variables 
  offspring_phenotype <- offspring_genetic + offspring_environment
  
  #creating PRSs 
  #PRSs, especially for psychiatric traits, are still poor measures of genetic variance due to two main reasons 
  #1) the amount of genetic variants measured (tagged genetic variance) is still very low - we capture this in our simulations with our varying tagged parameter
  #2) the estimates of the variant effect on the phenotype may not be accurate i.e. GWAS are still quite underpowered - we capture this with our varying V_noise parameter
  #Remember that the PRSs are purely derived from tagged genetic variance 
  V_noise = prs_noise
  
  #Creating parental PRSs where PRSs are derived of parental tagged genetic variables + noise derived using the noise variance variable created in the previous step
  maternal_noise = rnorm(n, 0, sd=sqrt(V_noise))
  maternal_prs = maternal_genetic_tagged + maternal_noise
  
  paternal_noise = rnorm(n, 0, sd=sqrt(V_noise)) 
  paternal_prs = paternal_genetic_tagged + paternal_noise 
  
  
  #Simulating offspring PRSs derived from offspring tagged genetic variable + noise derived from the average parental noise and offspring noise
  offspring_prs = offspring_genetic_tagged + rnorm(n, 0, sd=sqrt(V_noise/2)) + (maternal_noise + paternal_noise)/2
  
  
  #Creating a data.frame with all important variables included
  cohort <- data.frame(maternal_genetic, maternal_environment, maternal_phenotype, maternal_prs,
                       paternal_genetic, paternal_environment, paternal_phenotype, paternal_prs,
                       offspring_genetic, offspring_environment, offspring_phenotype, offspring_prs)
  return(cohort)}

  ########################################################
  #2. Save a dataframe for each scenario with 15 replicas#
  ######################################################## 

  for(replica in 1:15){
    for(h2 in c(0.3, 0.6, 0.9)){
      for(tagged in c(0.2, 0.6, 1)){
        for(prs_noise in seq(0,1,by=0.1)){
        
  cohort <- sim(h2, tagged, prs_noise)
  cohort$replica <- replica
  cohort$h2 <- h2
  cohort$tagged <- tagged
  cohort$prs_noise <- prs_noise
  
  write.table(cohort, file= paste0("cohort_h2", h2 , "_tagged", tagged, "_prs_noise", prs_noise), row.name=F, quote=F, col.names=F, append=T)
  }}}}

```

We expect 3x3x11 cohort files with 15,000 rows (15 replicas of 10,000) and 12 columns.  

Can check number of outputted rows using ``` tree  <directory name> ```  

Can count the rows of each file in the directory using ``` wc -l * ```

## ANALYSES USING SIMULATED DATASETS {.tabset}  

### Parallel Processing  

Making use of parallel processing in this case will ensure efficiency in compute time. 

For this process, a file of all potential combinations of the varying parameters (h2, tagged, prs_noise) need to be combined in a text file, where each row represents a scenario (e.g. 0.3, 0.2, 0)

```{r, eval=FALSE}

parameters <- c()

for(h2 in c(0.3, 0.6, 0.9)){
 for(tagged in c(0.2, 0.6, 1)){
   for(prs_noise in seq(0,1, by=0.1)){
   
     a <- c(h2, tagged, prs_noise)
     parameters <- rbind(parameters,a)
     
     }}}
               
write.table(parameters, "sim_no_IGE_parameters.txt", row.names=F, quote=F, col.names=F)
#this file will be used to specify the different datasets saved

```

Array job script

```
#!/bin/sh
###########################################

# M.Chuong, Edinburgh U.K, February 2021
# Eddie Indirect Genetic Effects Project Simulation Analysis

#$ -N sim_pa
#$ -t 1-99
#$ -tc 20
#$ -cwd
#$ -e e_o_outputs # place to save .e* error outputs
#$ -o e_o_outputs # place to save .o* error outputs
#$ -pe sharedmem 1 # number of cores
#$ -l h_vmem=2G # memory per core
#$ -l h_rt=24:00:00 ## requested time
#$ -m n ## notifications: (b)begin/(a)aborted/(e)end/(s)suspended/(n)nomail
#$ -M melisa.chuong@ed.ac.uk ## email for notifications


# INITIALISE ENV MODULES
. /etc/profile.d/modules.sh # if using modules need to add this line

# LOAD THE MODULES
module load igmm/apps/R/3.6.1

parameter_file=sim_noIGE_parameters.txt
line=`sed -n -e "${SGE_TASK_ID} p" ${parameter_file}`
parameters=($line)

h2=${parameters[0]}
tagged=${parameters[1]}
prs_noise=${parameters[2]}

Rscript sim_noIGE_pathway.R/sim_noIGE_regression.R $h2 $tagged $prs_noise

```
When running the array job what is happening is that each 'cohort' dataframe is identified using the parameters text file, and the following regression/pathway analyses are conducted. The array job allows for several cohort dataframes to run through the analyses simultaneously - as the actual regression/pathway analyses are identical.  

It is important that the results of each of these analyses are saved *separately* rather than appended to the same output file, even though this outputs a shitload of files (that can cause severe anxiety). Some analyses may fail and then the appended file will be all weird/ some analyses may complete at the same time (not common - BUT IT DOES HAPPEN, IT HAPPENED TO ME) and file format can corrupt as 2 output lines are being appended to the same row.  

So creating separate output files and then row binding them will work better (save yourself the tears).  

### Regression Simulations  

Save following as sim_noIGE_regression.R  
```{r, eval=FALSE}

#!/usr/bin/env Rscript
args = commandArgs(trailingOnly=TRUE)

h2 <- as.numeric(args[1])
tagged <- as.numeric(args[2])
prs_noise <- as.numeric(args[3])

sym= c()

df <- read.table(paste0("cohort_h2", h2, "_tagged", tagged, "_prs_noise", prs_noise), header=F)

colnames(df) <- c("maternal_genetic", "maternal_environment","maternal_phenotype",
                  "maternal_prs", "paternal_genetic",
                  "paternal_environment","paternal_phenotype", "paternal_prs",
                  "offspring_genetic","offspring_environment","offspring_phenotype",
                  "offspring_prs", "replica", "h2", "tagged", "prs_noise")

for(replica in 1:15){

cohort <- df[df$replica==replica,]

      regression <- summary(lm(offspring_phenotype ~ offspring_prs + maternal_prs + paternal_prs + maternal_phenotype + paternal_phenotype , data=cohort))
      reginfo <-  cbind(regression$coefficients, regression$adj.r.squared, prs_noise, replica, h2, tagged)  
      sym <- rbind(sym,reginfo) 
      }

write.csv(sym, paste0("sim_outcome_trioprs_parentpheno_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), row.names=F, quote=F)

```

Row bind all result output files

```{r, eval=FALSE}

      final <- as.data.frame(matrix(NA, nrow=1, ncol=10))
      colnames(final) <- c("Predictor", "Estimate", "SE", "T-value", "P-value", "AdjR2", "prs_noise", "replica", "h2", "Tagged")
      
      for(h2 in c(0.3, 0.6, 0.9)){
      for(tagged in c(0.2, 0.6, 1)){
      for(prs_noise in seq(0,1,by=0.1)){
        
      df <- read.csv(paste0("sim_outcome_trioprs_parentpheno_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), header=T)
      colnames(df) <- c("Predictor", "Estimate", "SE", "T-value", "P-value", "AdjR2", "prs_noise", "replica", "h2", "Tagged")
      final <- rbind(final, df)
      
      }}}

      write.table(final, "sim_noIGE_trioprs_parentpheno.txt", col.names=F, row.names=F, quote=F, append=T)   

```


### Pathway Model Simulations  

Save following as sim_noIGE_pathway.R
```{r, eval=FALSE}
#!/usr/bin/env Rscript
args = commandArgs(trailingOnly=TRUE)

h2 <- as.numeric(args[1])
tagged <- as.numeric(args[2])
prs_noise <- as.numeric(args[3])

library(lavaan)

sim_simple <- c()
sim_extended <- c()
sim_res_extended <- c()

df <- read.table(paste0("cohort_h2", h2, "_tagged", tagged, "_prs_noise", prs_noise), header=F)

colnames(df) <- c("maternal_genetic", "maternal_environment","maternal_phenotype",
                  "maternal_prs", "paternal_genetic",
                  "paternal_environment","paternal_phenotype", "paternal_prs",
                  "offspring_genetic","offspring_environment","offspring_phenotype",
                  "offspring_prs", "replica", "h2", "tagged", "prs_noise")

for(replica in 1:15){

cohort <- df[df$replica==replica,]
  
      ##############
      #SIMPLE MODEL#
      ##############
        
      # Define model
        
      model <- ' 
      
      #direct effects
      offspring_phenotype ~ c*maternal_prs
      offspring_phenotype ~ f*paternal_prs            
      
      #mediator
      offspring_prs ~ a*maternal_prs
      offspring_prs ~ d*paternal_prs
      offspring_phenotype ~ b*offspring_prs
      
      # indirect effects (a*b, d*b)
      ab := a*b
      db := d*b
      
      # total effects 
      total_mum := c + (a*b)
      total_dad := f + (d*b)
      '
        
      # Fit the model
      fitM <- sem(model, data=cohort, estimator='ML')
      
      sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
      sem_model_df <- as.matrix(sem_model$PE)
      sem_model_final <- sem_model_df[sem_model_df[,4] != "", c(1,3:4,6,7,9:11)]
      simple_model <- cbind(sem_model_final, AIC(fitM), replica, cohort$h2, cohort$tagged, cohort$prs_noise)
      
      sim_simple <- rbind(sim_simple, simple_model)
      
      
      ################
      #EXTENDED MODEL#
      ################
        
      # Define model
        
      model <- '

      #direct effects
      offspring_phenotype ~ c*maternal_prs
      offspring_phenotype ~ f*paternal_prs        

      #mediator
      offspring_prs ~ a*maternal_prs
      offspring_prs ~ d*paternal_prs 
      offspring_phenotype ~ b*offspring_prs
      paternal_phenotype ~ g*paternal_prs
      maternal_phenotype ~ h*maternal_prs
      offspring_phenotype ~ i*paternal_phenotype
      offspring_phenotype ~ j*maternal_phenotype
  
      # indirect effects (a*b, d*b)
      ab := a*b
      db := d*b
      gi := g*i
      hj := h*j
    
      # total effects 
      total_mum := c + (a*b) + (h*j)
      total_dad := f + (d*b) + (g*i)
    
      # covariance
      maternal_prs ~~ paternal_prs
      maternal_phenotype ~~ paternal_phenotype
      '
    
      # Fit the model
      fitM <- sem(model, data=cohort, estimator='ML')
      sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
      sem_model_df <- as.matrix(sem_model$PE)
      sem_model_final <- sem_model_df[sem_model_df[,4] != "", c(1,3:4,6,7,9:11)]
      extended_model <- cbind(sem_model_final, AIC(fitM), replica, h2, tagged, prs_noise)
    
      sim_extended <- rbind(sim_extended, extended_model)
      
      #############################
      #EXTENDED (RESTRICTED) MODEL#
      #############################
        
      # Define model
        
      model <- ' 

      #direct effects

      #mediator
      offspring_prs ~ a*maternal_prs
      offspring_prs ~ d*paternal_prs
      offspring_phenotype ~ b*offspring_prs
      paternal_phenotype ~ g*paternal_prs
      maternal_phenotype ~ h*maternal_prs
      offspring_phenotype ~ i*paternal_phenotype
      offspring_phenotype ~ j*maternal_phenotype

      # indirect effects (a*b, d*b)
      ab := a*b
      db := d*b
      gi := g*i
      hj := h*j


      # total effects 

      total_mum := (a*b) + (h*j) 
      total_dad := (d*b) + (g*i) 

      # covariance
      maternal_prs ~~ paternal_prs
      maternal_phenotype ~~ paternal_phenotype
      '
        
      # Fit the model
      fitM <- sem(model, data=cohort, estimator='ML')
      sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
      sem_model_df <- as.matrix(sem_model$PE)
      sem_model_final <- sem_model_df[sem_model_df[,4] != "", c(1,3:4,6,7,9:11)]
      extended_restricted_model <- cbind(sem_model_final, AIC(fitM), replica, h2, tagged, prs_noise)
	    sim_res_extended <- rbind(sim_res_extended, extended_restricted_model)
      }

write.csv(sim_simple, paste0("sim_pa_noIGE_m1_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), row.names=T, quote=F)
write.csv(sim_extended, paste0("sim_pa_noIGE_m2_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), row.names=T, quote=F)
write.csv(sim_res_extended, paste0("sim_pa_noIGE_m3_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), row.names=T, quote=F)

```

Row bind all result output files

```{r, eval=FALSE}

      #SIMPLE MODEL 

      final <- as.data.frame(matrix(NA, nrow=1, ncol=13))
      colnames(final) <- c("DV", "IV", "label", "Estimate", "SE", "P-value", "lowerCI", "upperCI", "AIC", "replica", "h2", "Tagged", "prs_noise")
      
      for(h2 in c(0.3, 0.6, 0.9)){
      for(tagged in c(0.2, 0.6, 1)){
      for(prs_noise in seq(0,1,by=0.1)){
        
      df <- read.csv(paste0("sim_pa_noIGE_m1_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), header=T)
      colnames(df) <- c("DV", "IV", "label", "Estimate", "SE", "P-value", "lowerCI", "upperCI", "AIC", "replica", "h2", "Tagged", "prs_noise")
                       
      final <- rbind(final, df)
      
      }}}

      write.table(final, "sim_pa_noIGE_m1.txt", col.names=F, row.names=F, quote=F, append=T)   
      
      #EXTENDED MODEL
      
      final <- as.data.frame(matrix(NA, nrow=1, ncol=13))
      colnames(final) <- c("DV", "IV", "label", "Estimate", "SE", "P-value", "lowerCI", "upperCI", "AIC", "replica", "h2", "Tagged", "prs_noise")
      
      for(h2 in c(0.3, 0.6, 0.9)){
      for(tagged in c(0.2, 0.6, 1)){
      for(prs_noise in seq(0,1,by=0.1)){
        
      df <- read.csv(paste0("sim_pa_noIGE_m2_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), header=T)
      colnames(df) <- c("DV", "IV", "label", "Estimate", "SE", "P-value", "lowerCI", "upperCI", "AIC", "replica", "h2", "Tagged", "prs_noise")
                       
      final <- rbind(final, df)
      
      }}}

      write.table(final, "sim_pa_noIGE_m2.txt", col.names=F, row.names=F, quote=F, append=T) 
      
      #EXTENDED RESTRICTED MODEL
      
      final <- as.data.frame(matrix(NA, nrow=1, ncol=13))
      colnames(final) <- c("DV", "IV", "label", "Estimate", "SE", "P-value", "lowerCI", "upperCI", "AIC", "replica", "h2", "Tagged", "prs_noise")
      
      for(h2 in c(0.3, 0.6, 0.9)){
      for(tagged in c(0.2, 0.6, 1)){
      for(prs_noise in seq(0,1,by=0.1)){
        
      df <- read.csv(paste0("sim_pa_noIGE_m3_h2", h2, "_tagged", tagged, "_prsnoise", prs_noise, ".csv"), header=T)
      colnames(df) <- c("DV", "IV", "label", "Estimate", "SE", "P-value", "lowerCI", "upperCI", "AIC", "replica", "h2", "Tagged", "prs_noise")
                       
      final <- rbind(final, df)
      
      }}}

      write.table(final, "sim_pa_noIGE_m3.txt", col.names=F, row.names=F, quote=F, append=T) 
```

## RESULT PLOTS  

**Regression Plots**

```{r, eval=FALSE}
library(ggplot2)

df <- read.table("sim_noIGE_trioprs_parentpheno.txt", header=F)
colnames(df) <- c("Predictor", "Estimate", "SE", "T-value", "P-value", "AdjR2", "prs_noise", "replica", "h2", "Tagged")
df <- df[-1,]

df$h2 <- as.factor(df$h2)
df$Tagged <- as.factor(df$Tagged)
df$prs_noise <- as.factor(df$prs_noise)

#offspring pgs estimates

df2 <- df[df$Predictor=="offspring_prs",]

#png(file="sim_noIGE_oprs.png")

image1 <- ggplot(data=df2, aes(x= prs_noise, y=Estimate, fill=Tagged)) +
  geom_boxplot(outlier.size=0) +
  scale_y_continuous(limits=c(0,1.1), breaks=seq(0.1,1, 0.1), expand = c(0, 0)) +
  xlab("Poylgenic Score (PGS) Noise") + 
  ylab("Offspring Pheno ~ Offspring PGS (Beta Estimate)") +
  facet_wrap(~h2) +
  theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold")) +
  ggtitle("Extended Regression Model Offspring PGS Beta Estimate ") +
  scale_fill_manual(values = c("pink", "lightblue", "red"))

image1
#dev.off()


#maternal pgs estimates

df2 <- df[df$Predictor=="maternal_prs",]

#png(file="sim_noIGE_mprs.png")

image2 <- ggplot(data=df2, aes(x= prs_noise, y=Estimate, fill=Tagged)) +
  geom_boxplot(outlier.size=0) +
  scale_y_continuous(limits=c(-0.5,0.5), breaks=seq(-0.5,0.5, 0.1), expand = c(0, 0)) +
  xlab("Poylgenic Score (PGS) Noise") + 
  ylab("Offspring Pheno ~ Maternal PGS (Beta Estimate)") +
  facet_wrap(~h2) +
  theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold")) +
  ggtitle("Extended Regression Model Maternal PGS Beta Estimate ") +
  scale_fill_manual(values = c("pink", "lightblue", "red"))
image2
#dev.off()

#paternal pgs estimates

df2 <- df[df$Predictor=="paternal_prs",]

#png(file="sim_noIGE_pprs.png")

image3 <- ggplot(data=df2, aes(x= prs_noise, y=Estimate, fill=Tagged)) +
  geom_boxplot(outlier.size=0) +
  scale_y_continuous(limits=c(-0.5,0.5), breaks=seq(-0.5,0.5, 0.1), expand = c(0, 0)) +
  xlab("Poylgenic Score (PGS) Noise") + 
  ylab("Offspring Pheno ~ Paternal PGS (Beta Estimate)") +
  facet_wrap(~h2) +
  theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold")) +
  ggtitle("Extended Regression Model Paternal PGS Beta Estimate ") +
  scale_fill_manual(values = c("pink", "lightblue", "red"))
image3
#dev.off()

#maternal pheno estimates

#png("sim_noIGE_mpheno.png")

df2 <- df[df$Predictor=="maternal_phenotype",]

image5 <- ggplot(data=df2, aes(x= prs_noise, y=Estimate, fill=Tagged)) +
  geom_boxplot(outlier.size=0) +
  scale_y_continuous(limits=c(-0.2,0.5), breaks=seq(-0.2,0.5, 0.1), expand = c(0, 0)) +
  xlab("Poylgenic Score (PGS) Noise") + 
  ylab("Offspring Pheno ~ Maternal Pheno (Beta Estimate)") +
  facet_wrap(~h2) +
  theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold")) +
  ggtitle("Extended Regression Model Maternal Pheno Beta Estimate ") +
  scale_fill_manual(values = c("pink", "lightblue", "red"))
image5
#dev.off()

#paternal pheno estimates

#png(file="sim_noIGE_ppheno.png")

df2 <- df[df$Predictor=="paternal_phenotype",]

image6 <- ggplot(data=df2, aes(x= prs_noise, y=Estimate, fill=Tagged)) +
  geom_boxplot(outlier.size=0) +
  scale_y_continuous(limits=c(-0.2,0.5), breaks=seq(-0.2,0.5, 0.1), expand = c(0, 0)) +
  xlab("Poylgenic Score (PGS) Noise") + 
  ylab("Offspring Pheno ~ Paternal Pheno (Beta Estimate)") +
  facet_wrap(~h2) +
  theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold")) +
  ggtitle("Extended Regression Model Paternal Pheno Beta Estimate ") +
  scale_fill_manual(values = c("pink", "lightblue", "red"))
image6
#dev.off()

#Adjusted R2

df2 <- df

#png(file="sim_noIGE_adjR2.png")

image7 <- ggplot(data=df2, aes(x= prs_noise, y=AdjR2, fill=Tagged)) +
  geom_boxplot(outlier.size=0) +
  scale_y_continuous(limits=c(0,1), breaks=seq(0,1,0.1), expand = c(0, 0)) +
  xlab("Poylgenic Score (PGS) Noise") + 
  ylab("Simulated Full Regression Model Accounted Variance") +
  facet_wrap(~h2) +
  theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold")) +
  ggtitle("Extended Regression Model Adjusted R2 ") +
  scale_fill_manual(values = c("pink", "lightblue", "red"))
image7
#dev.off()

```

**Pathway Plots**  

```{r, eval=FALSE}  

#here we are interested in labels c and f; the parental genetic nurturing pathways i.e. effect of parent prs on offpsring prs THAT DOES NOT GO through the offsprings own PRS or the parent pheno

df <- read.table("sim_pa_noIGE_m2.txt", header=F)
df <- df[-1,]
colnames(df) <- c("DV", "IV", "label", "Estimate", "SE", "P-value","upperCI", "lowerCI", "AIC", "replica", "h2", "Tagged", "prs_noise")

df$Tagged <- as.factor(df$Tagged)
df$h2 <- as.factor(df$h2)
df$prs_noise <- as.factor(df$prs_noise)

#maternal genetic nurturing

df2 <- df[df$label=="c",]

plot1 <- ggplot(data=df2, aes(x=prs_noise, y=Estimate, fill=Tagged)) + 
         geom_boxplot(outlier.size=0, fatten=1) + 
         scale_y_continuous(limits=c(-0.5,0.05), breaks=seq(-0.5,0.05, 0.1), expand = c(0, 0)) +
         xlab("Poylgenic Score (PGS) Noise") + 
         ylab("Offspring Pheno ~ Maternal PGS Beta Estimate") +
         facet_wrap(~h2) +
         theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold"), axis.text.x=element_text(angle=45)) +
         ggtitle("Extended Model Maternal Indirect Genetic Effect (Path C)") +
         scale_fill_brewer("Tagged Genetic Variance", palette="Accent") + 
         theme_bw()

ggsave(file="sim_mIGE_pathc.png", plot=plot1, width=15, height=6)
ggsave(file="sim_mIGE_pathc.svg", plot=plot1, width=15, height=6)

#paternal genetic nurturing

df3 <- df[df$label=="f",]

plot2 <- ggplot(data=df3, aes(x=prs_noise, y=Estimate, fill=Tagged)) +
         geom_boxplot(outlier.size=0, fatten=1) +
         scale_y_continuous(limits=c(-0.5,0.05), breaks=seq(-0.5,0.05, 0.1), expand = c(0, 0)) +
         xlab("Poylgenic Score (PGS) Noise") + 
         ylab("Offspring Pheno ~ Paternal PGS Beta Estimate") +
         facet_wrap(~h2) +
         theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold"), axis.text.x=element_text(angle=45)) +
         ggtitle("Extended Model Paternal Indirect Genetic Effect (Path F)") +
         scale_fill_brewer("Tagged Genetic Variance", palette="Accent") + 
         theme_bw()

ggsave(file="sim_pIGE_pathf.png", plot=plot2, width=15, height=6)
ggsave(file="sim_pIGE_pathf.svg", plot=plot2, width=15, height=6)

#maternal pheno

df4 <- df[df$label=="j",]

plot3 <- ggplot(data=df4, aes(x=prs_noise, y=Estimate, fill=Tagged)) +
         geom_boxplot(outlier.size=0, fatten=1) +
         scale_y_continuous(limits=c(-0.1,0.5), breaks=seq(-0.1,0.5, 0.1), expand = c(0, 0)) +
         xlab("Poylgenic Score (PGS) Noise") + 
         ylab("Offspring Pheno ~ Maternal Pheno Beta Estimate") +
         facet_wrap(~h2) +
         theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold"), axis.text.x=element_text(angle=45)) +
         ggtitle("Extended Model Offspring Pheno ~ Maternal Pheno Effect Beta Estimate (Path J)") +
         scale_fill_brewer("Tagged Genetic Variance", palette="Accent") + 
         theme_bw()

ggsave(file="sim_mIGE_pathj.png", plot=plot3, width=15, height=6)
ggsave(file="sim_mIGE_pathj.svg", plot=plot3, width=15, height=6)

#paternal pheno

df5 <- df[df$label=="i",]

plot4 <- ggplot(data=df5, aes(x=prs_noise, y=Estimate, fill=Tagged)) +
         geom_boxplot(outlier.size=0, fatten=1) +
         scale_y_continuous(limits=c(-0.1,0.5), breaks=seq(-0.1,0.5, 0.1), expand = c(0, 0)) +
         xlab("Poylgenic Score (PGS) Noise") + 
         ylab("Offspring Pheno ~ Paternal Pheno Beta Estimate") +
         facet_wrap(~h2) +
         theme(axis.text = element_text(size=10), strip.text = element_text(size=10), axis.title=element_text(size=10,face="bold")) +
         ggtitle("Extended Model Offspring Pheno ~ Paternal Pheno Effect Beta Estimate (Path I)") +
         scale_fill_brewer("Tagged Genetic Variance", palette="Accent") + 
         theme_bw()

ggsave(file="sim_pIGE_pathi.png", plot=plot4, width=15, height=6)
ggsave(file="sim_pIGE_pathi.svg", plot=plot4, width=15, height=6)

library(patchwork)

final_plot <- plot1 / plot3

ggsave(file="maternal_ige_plots.png", plot=final_plot, width=15, height=12)
ggsave(file="maternal_ige_plots.svg", plot=final_plot, width=15, height=12)

```
