---
title: "Genetic Nurturing Effects"
author: "Melisa Chuong"
date: "17/09/2022"
output: 
  html_document:
    code_folding: hide
---

# Motivation
# Generation Scotland Trios  

Genotyped data and pedigree files can be found on datastore @/exports/igmm/eddie/GenScotDepression/data/genscot/genetics/genotypes/GS20K_PLINK_files/QCd_data   
*These files are saved as GS_QC.bed/.bim/.fam*  

```{r fam, cache=TRUE}
fam <- read.table(here::here("data", "gs", "GS_QC.fam")) #this file was renamed for convenience 
colnames(fam) <- c("FID", "IID", "PID", "MID", "SEX", "PHENO")

fatherpresent <- fam[fam$PID%in%fam$IID,] #Finding PIDs also present in the IID column - every individual in the IID column in the pedigree file has genotyped data#
dim(fatherpresent) #fatherduo n= 3,865

motherpresent <- fam[fam$MID%in%fam$IID,] #Finding MIDs also present in the IID column#
dim(motherpresent) #motherduo n=5,954

trio <- fam[fam$PID%in%fam$IID & fam$MID%in%fam$IID,] #Finding trios, i.e. individuals with both PIDs and MIDs present in the IID column#
```

n= `r nrow(trio)` genotyped trios available in Generation Scotland

# Phenotypes {.tabset}  

## Covars

```{r covar, cache = TRUE}
age <- read.table(here::here("data", "gs", "GS_Ages.qcovar"), header=T)
age <- age[,c("IID", "age")]
colnames(age) <- c("IID", "AGE")

sex <- read.table(here::here("data", "gs", "GS_SexClinic.fcovar"), header=T)
sex <- sex[,c("IID", "sex")]
colnames(sex) <- c("IID", "SEX")

pcs <- read.table(here::here("data", "gs", "HM3mds.mds"), header=T)
pcs <- pcs[!pcs$FID %in% c("CEU", "CHB", "YRI", "TSI", "JPT", "CHD", 
"MEX", "GIH", "ASW", "LWK", "MKK"), ] # remove reference sets
pcs$FID <- NULL
pcs$SOL <- NULL

covars <- merge(age, sex, by="IID")
covars <- merge(covars, pcs, by="IID")

covar_formula <- . ~ AGE+SEX+C1+C2+C3+C4+C5+C6+C7+C8+C9+C10+C11+C12+C13+C14+C15+C16+C17+C18+C19+C20
```

## Height  

Raw Phenotype  

```{r height, cache=TRUE}

df <- load(here::here("data", "gs", "masterDB.Rdata"))
which(colnames(totaldata)=="height")

height <- totaldata[,c('id','height')]
height <- height[complete.cases(height),] #n=21,450
colnames(height) <- c("IID", "HEIGHT")
```

Trio Phenotype  

```{r trio_height, cache=TRUE}

height_covars <- merge(height, covars, "IID")

#Regressing out the effects of AGE, SEX, and PCs
height_cov_m <- lm(update(covar_formula, HEIGHT~.), data=height_covars)
height_res <- height_covars[, c("IID", "HEIGHT")]
height_res$HEIGHT_res <- residuals(height_cov_m)

summary(height_res$HEI_res)
sd(height_res$HEI_res)

trio_height <- merge(trio, height_res, by="IID", all.x=T)
trio_height <- merge(trio_height, height_res, by.x = "MID", by.y = "IID", suffixes = c('', '_mat'))
trio_height <- merge(trio_height, height_res, by.x = "PID", by.y = "IID", suffixes = c('', '_pat'))

```

Phenotype Demographics  

```{r height_demo}
trio_height_complete <- trio_height[complete.cases(trio_height),]
pheno_info <- c()

for(column in c('HEIGHT', 'HEIGHT_mat', 'HEIGHT_pat')){
  min_pheno <- min(trio_height_complete[column])
  max_pheno <- max(trio_height_complete[column])
  mean_pheno <- sum(trio_height_complete[column])/dim(trio_height_complete)[1]
  height_info <- Reduce(function(x,y) cbind(x=x, y=y), list(column, min_pheno, mean_pheno, max_pheno, dim(trio_height_complete)[1]))
  pheno_info <- rbind(pheno_info, height_info)
}

```

## Educational Attainment  

Raw Phenotype  

```{r ea, cache=TRUE}

ea <- totaldata[,c("id", "years_category")]
colnames(ea) <- c("IID", "EA")
ea <- ea[complete.cases(ea),]

```

Trios Phenotype  

```{r ea_trio}

ea_covars <- merge(ea, covars, by = "IID")
ea_covars <- ea_covars[complete.cases(ea_covars),]

#Regressing out the effects of AGE and SEX
ea_cov_m <- lm(update(covar_formula, EA~.), data=ea_covars)
ea_res <- ea_covars[, c("IID", "EA")]
ea_res$EA_res <- residuals(ea_cov_m)

summary(ea_res$EA_res)
sd(ea_res$EA_res)

trio_ea <- merge(trio, ea_res, by="IID", all.x=T)
trio_ea <- merge(trio_ea, ea_res, by.x = "MID", by.y = "IID", suffixes = c('', '_mat'))
trio_ea <- merge(trio_ea, ea_res, by.x = "PID", by.y = "IID", suffixes = c('', '_pat'))


```

Phenotype Demographics  

```{r, eval=FALSE}
ea_complete <- trio_ea[complete.cases(trio_ea),]

for(column in c('EA', 'EA_mat', 'EA_pat')){
  min_pheno <- min(ea_complete[column])
  max_pheno <- max(ea_complete[column])
  mean_pheno <- sum(ea_complete[column])/dim(ea_complete)[1]
  ea_info <- Reduce(function(x,y) cbind(x=x, y=y), list(column, min_pheno, mean_pheno, max_pheno, dim(ea_complete)[1]))
  pheno_info <- rbind(pheno_info, ea_info)
}

```

## Depressive symptoms 

Raw Phenotype  

```{r ghq}

ghq <- read.csv(here::here("data", "gs", "GS_All_GHQ.csv"))
ghq_dep <- ghq[,c("id", "GHQDepression")]
ghq_dep <- ghq_dep[complete.cases(ghq_dep),]
names(ghq_dep) <- c("IID", "GHQD")


```

Trios Phenotype  

```{r ghq_trios}

ghqd_covars <- merge(ghq_dep, covars, by=c("IID")) #n=19,961

#Regressing out the effects of AGE and SEX
ghqd_cov_m <- lm(update(covar_formula, GHQD~.), data=ghqd_covars)
ghqd_res <- ghqd_covars[, c("IID", "GHQD")]
ghqd_res$GHQD_res <- residuals(ghqd_cov_m)

summary(ghqd_res$GHQD_res)
sd(ghqd_res$GHQD_res)

trio_ghqd <- merge(trio, ghqd_res, by="IID", all.x=T)
trio_ghqd <- merge(trio_ghqd, ghqd_res, by.x = "MID", by.y = "IID", suffixes = c('', '_mat'))
trio_ghqd <- merge(trio_ghqd, ghqd_res, by.x = "PID", by.y = "IID", suffixes = c('', '_pat'))

```

Phenotype Demographics  

```{r ghq_demo}
trio_ghqd_compete <- trio_ghqd[complete.cases(trio_ghqd),]

for(column in c('GHQD', 'GHQD_mat', 'GHQD_pat')){
  min_pheno <- min(trio_ghqd_compete[column])
  max_pheno <- max(trio_ghqd_compete[column])
  mean_pheno <- sum(trio_ghqd_compete[column])/dim(trio_ghqd_compete)[1]
  ghqt_info <- Reduce(function(x,y) cbind(x=x, y=y), list(column, min_pheno, mean_pheno, max_pheno, dim(trio_ghqd_compete)[1]))
  pheno_info <- rbind(pheno_info, ghqt_info)
}

```

# Polygenic Risk Scores {.tabset}  

GenoPred (Pain, Al-Chalabi, & Lewis, 2024) is used to generate the PRSs using SBayesR (Lloyd-Jones 2019) and unweighted clumping-and-thresholding.  
Phenotypes will be corrected for covariates.  
We will use 1000G as a reference panel for LD estimation which will go onto guide pruning and clumping of SNPs. 

We will use:

Base data:  
Height GWAS meta-analysis sumstats  ~ Yengo et al., 2018: Wood et al., 2014 N= 253,288 individuals (NO GENSCOT) + UKB (Sample size of 693,529)  

EA GWAS meta-analysis sumstats  ~ Lee et al., 2018 (removing 23&Me + GS data) N= 1,131,880-381,477 =750,403   

Depression GWAS sumstats ~ Howard et al., 2019, (removing GS) (ncase=242,956; ncontrol=542,391)   

```{r prs}  

hPRS_sbayesr <- read.table(here::here("data/pgs/sbayesr/Height_Yengo2018/genscot-Height_Yengo2018-EUR.profiles"), header = TRUE)
eaPRS_sbayesr <- read.table(here::here("data/pgs/sbayesr/EduYears_Lee2018/genscot-EduYears_Lee2018-EUR.profiles"), header = TRUE)
depPRS_sbayesr <- read.table(here::here("data/pgs/sbayesr/MDD_Howard2019/genscot-MDD_Howard2019-EUR.profiles"), header = TRUE)


hPRS_unweighted <- read.table(here::here("data/pgs/ptclump/Height_Yengo2018_unweighted/genscot-Height_Yengo2018_unweighted-EUR.profiles"), header = TRUE)
eaPRS_unweighted <- read.table(here::here("data/pgs/ptclump/EduYears_Lee2018_unweighted/genscot-EduYears_Lee2018_unweighted-EUR.profiles"), header = TRUE)
depPRS_unweighted <- read.table(here::here("data/pgs/ptclump/MDD_Howard2019_unweighted/genscot-MDD_Howard2019_unweighted-EUR.profiles"), header = TRUE)

PRS <- merge(hPRS_sbayesr, eaPRS_sbayesr, by = c("FID", "IID"))
PRS <- merge(PRS, depPRS_sbayesr, by = c("FID", "IID"))
PRS <- merge(PRS, hPRS_unweighted, by = c("FID", "IID"))
PRS <- merge(PRS, eaPRS_unweighted, by = c("FID", "IID"))
PRS <- merge(PRS, depPRS_unweighted, by = c("FID", "IID"))
PRS$FID <- NULL

```

# Feature Checks  

**1. Are the calculated PGS_pcs associated with our phenotypes of interest?**  

```{r checks, cache = TRUE}

library(lme4)
library(ggplot2)

phenos <- merge(height, ea, by = "IID", all = TRUE)
phenos <- merge(phenos, ghq_dep, by = "IID", all = TRUE)

phenos_prs <- merge(phenos, PRS, by = "IID")
phenos_prs <- merge(phenos_prs, fam, by = "IID")

phenos_prs_map <- c("HEIGHT" = "Height_Yengo2018",
                    "EA" = "EduYears_Lee2018",
                    "GHQD" = "MDD_Howard2019")
prs_methods <- c("SBayesR", paste0("unweighted_0_", c("5e.08", "1e.06", "1e.04", "0.01", "0.1", "0.2", "0.3", "0.4", "0.5", "1")))

pheno_prs_checks <- data.frame(pheno = rep(names(phenos_prs_map), times = length(prs_methods)),
                               method = rep(prs_methods, each = length(names(phenos_prs_map))),
                               beta = NA, se = NA)
                               
for(i in seq.int(nrow(pheno_prs_checks))) {
  pheno <- pheno_prs_checks[i,"pheno"]
  method <- pheno_prs_checks[i,"method"]

  prs_predictor <- paste0(phenos_prs_map[[pheno]], "_", method)
  prs_formula <- as.formula(paste(pheno, "~", prs_predictor))
  m <- lmer(update(prs_formula, . ~ . + (1 | FID)), data = phenos_prs)
  m_summary <- summary(m)$coefficients
  b <- m_summary[prs_predictor,'Estimate']
  se <- m_summary[prs_predictor,'Std. Error']
  
  sd_predictor <- sd(phenos_prs[,prs_predictor], na.rm=T)
  sd_outcome <- sd(phenos_prs[,pheno], na.rm=T)
  
  b_std <- b * sd_predictor / sd_outcome
  se_std <- se * sd_predictor / sd_outcome
  
  pheno_prs_checks[i,"beta"] <- b_std
  pheno_prs_checks[i,"se"] <- se_std
  
}

ggplot(pheno_prs_checks, aes(x = method, y = beta, ymin = beta + qnorm(0.025, sd=se), ymax = beta + qnorm(0.975, sd=se))) + geom_pointrange() + facet_grid(~pheno) + coord_flip()

  
```

**2. Are the trio PGS_pcs and phenotypes correlated in the way we would expect?**  

```{r trio_prs, cache = TRUE}

# TRIO PGS correlations

trio_prs <- merge(trio, PRS, by="IID", all.x=T)
trio_prs <- merge(trio_prs, PRS, by.x = "MID", by.y = "IID", suffixes = c('', '_mat'))
trio_prs <- merge(trio_prs, PRS, by.x = "PID", by.y = "IID", suffixes = c('', '_pat'))


prs_mpo_cor <- data.frame(pheno = rep(names(phenos_prs_map), times = length(prs_methods)),
                               method = rep(prs_methods, each = length(names(phenos_prs_map))),
                               rMO = NA, rPO = NA, rMP = NA,
                               rMOlower = NA, rPOlower = NA, rMPlower = NA,
                               rMOupper = NA, rPOupper = NA, rMPupper = NA)
                               
for(i in seq.int(nrow(pheno_prs_checks))) {
  pheno <- pheno_prs_checks[i,"pheno"]
  method <- pheno_prs_checks[i,"method"]

  prs_predictor <- paste0(phenos_prs_map[[pheno]], "_", method)
  prs_predictor_m <- paste0(prs_predictor, "_mat")
  prs_predictor_p <- paste0(prs_predictor, "_pat")
  
  rMO <- cor.test(trio_prs[,prs_predictor], trio_prs[,prs_predictor_m])
  rPO <- cor.test(trio_prs[,prs_predictor], trio_prs[,prs_predictor_p])
  rMP <- cor.test(trio_prs[,prs_predictor_m], trio_prs[,prs_predictor_p])
   
  prs_mpo_cor[i,"rMO"] <- rMO$estimate[['cor']]
  prs_mpo_cor[i,"rPO"] <- rPO$estimate[['cor']]
  prs_mpo_cor[i,"rMP"] <- rMP$estimate[['cor']]
   
  prs_mpo_cor[i,"rMOlower"] <- rMO$conf.int[1]
  prs_mpo_cor[i,"rPOlower"] <- rPO$conf.int[1]
  prs_mpo_cor[i,"rMPlower"] <- rMP$conf.int[1] 
  
  prs_mpo_cor[i,"rMOupper"] <- rMO$conf.int[2]
  prs_mpo_cor[i,"rPOupper"] <- rPO$conf.int[2]
  prs_mpo_cor[i,"rMPupper"] <- rMP$conf.int[2] 
}

# reshape
prs_mo_cor <- prs_mpo_cor[,c("pheno", "method", "rMO", "rMOlower", "rMOupper")]
prs_mo_cor$est <- "rMO"
prs_po_cor <- prs_mpo_cor[,c("pheno", "method", "rPO", "rPOlower", "rPOupper")]
prs_po_cor$est <- "rPO"
prs_mp_cor <- prs_mpo_cor[,c("pheno", "method", "rMP", "rMPlower", "rMPupper")]
prs_mp_cor$est <- "rMP"

names(prs_mo_cor)[3:5] <- names(prs_po_cor)[3:5] <- names(prs_mp_cor)[3:5] <- c("r", "r_upper", "r_lower")
prs_mp_cor_long <- rbind(prs_mo_cor, prs_po_cor, prs_mp_cor)

ggplot(prs_mp_cor_long, aes(x = method, y = r, ymin = r_lower, ymax = r_upper, color = est)) +
  geom_hline(yintercept = 0, colour = 'darkgray') +
  geom_hline(yintercept = 0.5, colour = 'darkgray') +
  geom_pointrange(position = position_dodge(width=0.3)) + 
  facet_grid(~pheno) + 
  coord_flip()

```

# Analysis Dataframe  

PRS are scaled to the 1000G EUR reference group..  
All variables are pre-corrected for covariates, 20 offspring PCs, 20 maternal and paternal PCs, parental age, offspring age, offspring sex. 

# Regression Analyses  

*I*. oPheno ~ oPRS *This model captures direct GE*  
*II*. oPheno ~ oPRS + mPRS + pPRS *This model captures both direct and indirect GE*   
*III*. oPheno ~ oPRS + mPheno + pPheno  
*IV*. oPheno ~ oPRS + mPheno + pPheno+ mPRS + pPRS  

*We are not interpreting beta estimates as these are likely to be biased/inaccurate as variables in the models are correlated!*
Comparison of the models using log likelihood can help us infer whether the effects of indirect genetic effects are present (specifically comparison of model *I* and *II*)  

```{r reg_models, eval=FALSE}
library(lmerTest) #this package allows us to run regression models with fixed and random effects
library(MuMIn) #this package allows us to obtain the R2 of our models, R2m (Marginal) R2 > fixed effects, R2c (Conditional) > whole model
library(dplyr)

trio_phenos <- merge(trio_height, trio_ea, by = c("IID", "MID", "PID", "FID", "SEX", "PHENO"))
trio_phenos <- merge(trio_phenos, trio_ghqd, by = c("IID", "MID", "PID", "FID", "SEX", "PHENO"))

trio_phenos_prs <- merge(trio_phenos, trio_prs, by = c("IID", "MID", "PID", "FID", "SEX", "PHENO"))
trio_phenos_prs$sibID <- with(trio_phenos_prs, paste(MID, PID, sep = "_"))

trio_phenos_prs_complete <- trio_phenos_prs[complete.cases(trio_phenos_prs),]

phenos_prs_params <- list()

for(pheno in c("HEIGHT", "EA", "GHQD")) {
  for(method in c("SBayesR", "unweighted_0_1e.04")) {
    oPheno <- paste0(pheno, "_res")
    oPRS <- paste0(phenos_prs_map[[pheno]], '_', method)
    params <- list(pheno = pheno, method = method,
         oPheno = oPheno,
         mPheno = paste0(oPheno, '_mat'),
         pPheno = paste0(oPheno, '_pat'),
         oPRS = oPRS,
         mPRS = paste0(oPRS, '_mat'),
         pPRS = paste0(oPRS, '_pat'))
         
    phenos_prs_params <- c(phenos_prs_params, list(params))
    }
}

params_model1_formulae <- lapply(phenos_prs_params, function(params) {
  params$model <- 'model1'
  params$formula <- as.formula(paste(params$oPheno, ' ~ ', params$oPRS, '+ (1|sibID)'))
  return(params)
})

params_model2_formulae <- lapply(params_model1_formulae, function(params) {
  params$model <- 'model2'
  params$formula <- update(params$formula, as.formula(paste('. ~ . + ', params$mPRS, ' + ', params$pPRS)))
  return(params)
})

params_model3_formulae <- lapply(params_model1_formulae, function(params) {
  params$model <- 'model3'
  params$formula <- update(params$formula, as.formula(paste('. ~ . + ', params$mPheno, ' + ', params$pPheno)))
  return(params)
})

params_model4_formulae <- lapply(params_model2_formulae, function(params) {
  params$model <- 'model4'
  params$formula <- update(params$formula, as.formula(paste('. ~ . + ', params$mPheno, ' + ', params$pPheno)))
  return(params)
})

params_model_formulae <- c(params_model1_formulae, params_model2_formulae, params_model3_formulae, params_model4_formulae)


```


```{r, eval = FALSE}

=======

df <- read.table("IGE_analysis_df", header=T)
df$sibID <- paste0(df$MID, '_', df$PID)
df$index <- rownames(df)

for(pheno_prefix in c('HEIGHT', 'EA', 'GHQTotal', 'GHQDepression')){
  
  if(pheno_prefix=='HEIGHT'){
    prs_prefix <- 'h'
    oPhenoCol <- 5
    mPhenoCol <- 7
    pPhenoCol <- 9
    oPRScol <- 6
    mPRScol <- 8
    pPRScol <- 10
    
  } else if (pheno_prefix=='EA'){
    prs_prefix <- 'ea'
    oPhenoCol <- 11
    mPhenoCol <- 13
    pPhenoCol <- 15
    oPRScol <- 12
    mPRScol <- 14
    pPRScol <- 16
    
  } else if (pheno_prefix=='GHQTotal'){
    prs_prefix <- 'dep'
    oPhenoCol <- 17
    mPhenoCol <- 18
    pPhenoCol <- 19
    oPRScol <- 21
    mPRScol <- 23
    pPRScol <- 25
    
  } else {
    prs_prefix <- 'dep'
    oPhenoCol <- 20
    mPhenoCol <- 22
    pPhenoCol <- 24
    oPRScol <- 21
    mPRScol <- 23
    pPRScol <- 25
  }
  
  # all trios
  alldf_index <- df[, c(oPhenoCol, mPhenoCol, pPhenoCol, oPRScol, mPRScol, pPRScol, 28)]
  alldf_index <- as.numeric(alldf_index[complete.cases(alldf_index),7])
  alldf <- df[rownames(df)%in%alldf_index,]
  
  model1 <- lmer(alldf[,oPhenoCol] ~ alldf[,oPRScol] + (1|sibID), data = alldf)
  results_model1 <- cbind(as.data.frame(coef(summary(model1))), as.data.frame(r.squaredGLMM(model1)))
  rownames(results_model1) <- c('Intercept', 'oPRS')
  
  model2 <- lmer(alldf[,oPhenoCol] ~ alldf[,oPRScol] + alldf[,mPRScol] + alldf[,pPRScol] + (1|sibID), data = alldf)
  results_model2 <- cbind(as.data.frame(coef(summary(model2))), as.data.frame(r.squaredGLMM(model2)))
  rownames(results_model2) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS')

  model3 <- lmer(alldf[,oPhenoCol] ~ alldf[,oPRScol] + alldf[,mPhenoCol] + alldf[,pPhenoCol] + (1|sibID), data = alldf)
  results_model3 <- cbind(as.data.frame(coef(summary(model3))), as.data.frame(r.squaredGLMM(model3)))
  rownames(results_model3) <- c('Intercept', 'oPRS', 'mPheno', 'pPheno')
  
  model4 <- lmer(alldf[,oPhenoCol] ~ alldf[,oPRScol] + alldf[,mPRScol] + alldf[,pPRScol] + alldf[,mPhenoCol] + alldf[,pPhenoCol] + (1|sibID), data = alldf)
  results_model4 <- cbind(as.data.frame(coef(summary(model4))), as.data.frame(r.squaredGLMM(model4)))
  rownames(results_model4) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS', 'mPheno', 'pPheno')
  
  results <- Reduce(function(x,y) rbind(x=x, y=y), list(results_model1, results_model2, results_model3, results_model4))
  
  write.csv(results, paste0(pheno_prefix, 'regression_results.csv'), row.names=T, quote=F)
  
  # comparison
  
  model1_v_2 <- as.data.frame(anova(model1, model2))
  model1_v_2$comparison <- 'model1_v_2'
  
  model1_v_3 <- as.data.frame(anova(model1, model3))
  model1_v_3$comparison <- 'model1_v_3'
  
  model1_v_4 <- as.data.frame(anova(model1, model4))
  model1_v_4$comparison <- 'model1_v_4'
  
  model2_v_4 <- as.data.frame(anova(model2, model4))
  model2_v_4$comparison <- 'model2_v_4'
  
  model3_v_4 <- as.data.frame(anova(model3, model4))
  model3_v_4$comparison <- 'model3_v_4'
  
  results <- bind_rows(model1_v_2, model1_v_3, model1_v_4, model2_v_4, model3_v_4)
  results$fdr <- p.adjust(results$`Pr(>Chisq)`, method="fdr", n=length(results$`Pr(>Chisq)`))
  
  write.csv(results, paste0(pheno_prefix, 'regression_model_comparison.csv'), row.names=F, quote=F)
  
  # sex specific  
  for(sex_prefix in c('M', 'F')){
    
    if(sex_prefix=='M'){
      sex <- 1
    } else {
      sex <- 2
    }
    
    sexdf_index <- subset(alldf, alldf$SEX==sex)
    sexdf_index <- as.numeric(sexdf_index[,28])
    sexdf <- df[rownames(df)%in%sexdf_index,]
    
    model1 <- lmer(sexdf[,oPhenoCol] ~ sexdf[,oPRScol] + (1|sibID), data = sexdf)
    results_model1 <- cbind(as.data.frame(coef(summary(model1))), as.data.frame(r.squaredGLMM(model1)))
    rownames(results_model1) <- c('Intercept', 'oPRS')
    
    model2 <- lmer(sexdf[,oPhenoCol] ~ sexdf[,oPRScol] + sexdf[,mPRScol] + sexdf[,pPRScol] + (1|sibID), data = sexdf)
    results_model2 <- cbind(as.data.frame(coef(summary(model2))), as.data.frame(r.squaredGLMM(model2)))
    rownames(results_model2) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS')
  
    model3 <- lmer(sexdf[,oPhenoCol] ~ sexdf[,oPRScol] + sexdf[,mPhenoCol] + sexdf[,pPhenoCol] + (1|sibID), data = sexdf)
    results_model3 <- cbind(as.data.frame(coef(summary(model3))), as.data.frame(r.squaredGLMM(model3)))
    rownames(results_model3) <- c('Intercept', 'oPRS', 'mPheno', 'pPheno')
    
    model4 <- lmer(sexdf[,oPhenoCol] ~ sexdf[,oPRScol] + sexdf[,mPRScol] + sexdf[,pPRScol] + sexdf[,mPhenoCol] + sexdf[,pPhenoCol] + (1|sibID), data = sexdf)
    results_model4 <- cbind(as.data.frame(coef(summary(model4))), as.data.frame(r.squaredGLMM(model4)))
    rownames(results_model4) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS', 'mPheno', 'pPheno')
    
    results <- Reduce(function(x,y) rbind(x=x, y=y), list(results_model1, results_model2, results_model3, results_model4))
    
    write.csv(results, paste0(pheno_prefix, 'regression_results_', sex_prefix, '.csv'), row.names=T, quote=F)
    
  }
  
  # singleton analyses  
  
  singledf_index <- alldf[!duplicated(alldf$sibID),]
  singledf_index <- as.numeric(singledf_index[,28])
  singledf <- df[rownames(df)%in%singledf_index,]
  
  if(pheno_prefix=='HEIGHT' | pheno_prefix=='EA'){
    
    model1 <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol], data = singledf)
    results_model1 <- cbind(as.data.frame(summary(model1)$coefficients), as.data.frame(summary(model1)$adj.r.squared))
    rownames(results_model1) <- c('Intercept', 'oPRS')
    colnames(results_model1)[5] <- 'adjR2'
      
    model2 <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPRScol] + singledf[,pPRScol], data = singledf)
    results_model2 <- cbind(as.data.frame(summary(model2)$coefficients), as.data.frame(summary(model2)$adj.r.squared))
    rownames(results_model2) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS')
    colnames(results_model2)[5] <- 'adjR2'
    
    model3 <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPhenoCol] + singledf[,pPhenoCol], data = singledf)
    results_model3 <- cbind(as.data.frame(summary(model3)$coefficients), as.data.frame(summary(model3)$adj.r.squared))
    rownames(results_model3) <- c('Intercept', 'oPRS', 'mPheno', 'pPheno')
    colnames(results_model3)[5] <- 'adjR2'

    model4 <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPRScol] + singledf[,pPRScol] + singledf[,mPhenoCol] + singledf[,pPhenoCol], data = singledf)
    results_model4 <- cbind(as.data.frame(summary(model4)$coefficients), as.data.frame(summary(model4)$adj.r.squared))
    rownames(results_model4) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS', 'mPheno', 'pPheno')
    colnames(results_model4)[5] <- 'adjR2'

    results <- Reduce(function(x,y) rbind(x=x, y=y), list(results_model1, results_model2, results_model3, results_model4))
    
  } else {
    
    model1 <- glm(singledf[,oPhenoCol] ~ singledf[,oPRScol], data = singledf)
    model1_lm <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol], data = singledf)
    results_model1 <- cbind(as.data.frame(summary(model1)$coefficients), as.data.frame(summary(model1_lm)$adj.r.squared))
    rownames(results_model1) <- c('Intercept', 'oPRS')
    colnames(results_model1)[5] <- 'adjR2'
    
    model2 <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPRScol] + singledf[,pPRScol], data = singledf)
    model2_lm <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPRScol] + singledf[,pPRScol], data = singledf)
    results_model2 <- cbind(as.data.frame(summary(model2)$coefficients), as.data.frame(summary(model2_lm)$adj.r.squared))
    rownames(results_model2) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS')
    colnames(results_model2)[5] <- 'adjR2'

    model3 <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPhenoCol] + singledf[,pPhenoCol], data = singledf)
    model3_lm <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPhenoCol] + singledf[,pPhenoCol], data = singledf)
    results_model3 <- cbind(as.data.frame(summary(model3)$coefficients), as.data.frame(summary(model3_lm)$adj.r.squared))
    rownames(results_model3) <- c('Intercept', 'oPRS', 'mPheno', 'pPheno')
    colnames(results_model3)[5] <- 'adjR2'
      
    model4 <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPRScol] + singledf[,pPRScol] + singledf[,mPhenoCol] + singledf[,pPhenoCol], data = singledf)
    model4_lm <- lm(singledf[,oPhenoCol] ~ singledf[,oPRScol] + singledf[,mPRScol] + singledf[,pPRScol] + singledf[,mPhenoCol] + singledf[,pPhenoCol], data = singledf)
    results_model4 <- cbind(as.data.frame(summary(model4)$coefficients), as.data.frame(summary(model4_lm)$adj.r.squared))
    rownames(results_model4) <- c('Intercept', 'oPRS', 'mPRS', 'pPRS', 'mPheno', 'pPheno')
    colnames(results_model4)[5] <- 'adjR2'

    results <- Reduce(function(x,y) rbind(x=x, y=y), list(results_model1, results_model2, results_model3, results_model4))
  }
  
  write.csv(results, paste0(pheno_prefix, 'regression_results_singleton.csv'), row.names=T, quote=F)
}

# problem is, oPhenoCol are not aligning with actual col numbers 
# eg alldf oPhenoCol should be col 1

```

# Pathway models {.tabset}

Create separate dataframes for each phenotype

```{r, eval=FALSE}

rm(list=ls())
library(lavaan)

df <- read.table("IGE_analysis_df", header=T)
df$sibID <- paste0(df$MID, '_', df$PID)
df$index <- rownames(df)

for(pheno_prefix in c('HEIGHT', 'EA', 'GHQTotal', 'GHQDepression')){
  
  if(pheno_prefix=='HEIGHT'){
    prs_prefix <- 'h'
    oPhenoCol <- 5
    mPhenoCol <- 7
    pPhenoCol <- 9
    oPRScol <- 6
    mPRScol <- 8
    pPRScol <- 10
    
  } else if (pheno_prefix=='EA'){
    prs_prefix <- 'ea'
    oPhenoCol <- 11
    mPhenoCol <- 13
    pPhenoCol <- 15
    oPRScol <- 12
    mPRScol <- 14
    pPRScol <- 16
    
  } else if (pheno_prefix=='GHQTotal'){
    prs_prefix <- 'dep'
    oPhenoCol <- 17
    mPhenoCol <- 18
    pPhenoCol <- 19
    oPRScol <- 21
    mPRScol <- 23
    pPRScol <- 25
    
  } else {
    prs_prefix <- 'dep'
    oPhenoCol <- 20
    mPhenoCol <- 22
    pPhenoCol <- 24
    oPRScol <- 21
    mPRScol <- 23
    pPRScol <- 25
  }
  
  # create pathway dataframes 
  triodf <- df[, c(oPhenoCol, mPhenoCol, pPhenoCol, oPRScol, mPRScol, pPRScol, 26:28)]
  triodf <- triodf[complete.cases(triodf),]
  
  write.csv(triodf, paste0(pheno_prefix, "_pathwaymodel_df.csv"), row.names=F, quote=F)
}

```

## Height  

Path h & i restricted to 0.5:  

```{r, eval=FALSE}

rm(list=ls())
library(lavaan)

df <- read.csv("HEIGHT_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres
           
# mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres
HEIGHT_res ~ a*h_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_h_PRSres ~~ p_h_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "height_prspc_pa_m1.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres       

#mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres 

HEIGHT_res ~ a*h_PRSres

p_HEIGHT_res ~ d*p_h_PRSres
m_HEIGHT_res ~ e*m_h_PRSres

HEIGHT_res ~ f*p_HEIGHT_res
HEIGHT_res ~ g*m_HEIGHT_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_h_PRSres ~~ p_h_PRSres
m_HEIGHT_res ~~ p_HEIGHT_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "height_prspc_pa_m2.csv", row.names=F, quote=F)

```

All paths freely estimated:

```{r, eval=FALSE}

rm(list=ls())
library(lavaan)

df <- read.csv("HEIGHT_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres
           
# mediator
h_PRSres ~ i*m_h_PRSres
h_PRSres ~ h*p_h_PRSres
HEIGHT_res ~ a*h_PRSres

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a

# total effects 
total_mum := c + (i*a)
total_dad := b + (h*a)

# covariance
m_h_PRSres ~~ p_h_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "height_prspc_pa_m1_fm.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres       

#mediator
h_PRSres ~ i*m_h_PRSres
h_PRSres ~ h*p_h_PRSres 

HEIGHT_res ~ a*h_PRSres

p_HEIGHT_res ~ d*p_h_PRSres
m_HEIGHT_res ~ e*m_h_PRSres

HEIGHT_res ~ f*p_HEIGHT_res
HEIGHT_res ~ g*m_HEIGHT_res

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (i*a) + (e*g) 
total_dad := b + (h*a) + (d*f) 

# covariance
m_h_PRSres ~~ p_h_PRSres
m_HEIGHT_res ~~ p_HEIGHT_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "height_prspc_pa_m2_fm.csv", row.names=F, quote=F)

```

Female only:  

```{r, eval=FALSE}

rm(list=ls())
library(lavaan)

df <- read.csv("HEIGHT_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==2)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres
           
# mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres
HEIGHT_res ~ a*h_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_h_PRSres ~~ p_h_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "height_prspc_pa_m1_F.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres       

#mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres 

HEIGHT_res ~ a*h_PRSres

p_HEIGHT_res ~ d*p_h_PRSres
m_HEIGHT_res ~ e*m_h_PRSres

HEIGHT_res ~ f*p_HEIGHT_res
HEIGHT_res ~ g*m_HEIGHT_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_h_PRSres ~~ p_h_PRSres
m_HEIGHT_res ~~ p_HEIGHT_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "height_prspc_pa_m2_F.csv", row.names=F, quote=F)

```

Male only: 

```{r, eval=FALSE}
rm(list=ls())
library(lavaan)

df <- read.csv("HEIGHT_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==1)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres
           
# mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres
HEIGHT_res ~ a*h_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_h_PRSres ~~ p_h_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "height_prspc_pa_m1_M.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres       

#mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres 

HEIGHT_res ~ a*h_PRSres

p_HEIGHT_res ~ d*p_h_PRSres
m_HEIGHT_res ~ e*m_h_PRSres

HEIGHT_res ~ f*p_HEIGHT_res
HEIGHT_res ~ g*m_HEIGHT_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_h_PRSres ~~ p_h_PRSres
m_HEIGHT_res ~~ p_HEIGHT_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "height_prspc_pa_m2_M.csv", row.names=F, quote=F)

```

Singleton:

```{r, eval=FALSE}
rm(list=ls())
library(lavaan)

df <- read.csv("HEIGHT_pathwaymodel_df.csv", header=T)
df <- df[!duplicated(df$sibID),]

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres
           
# mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres
HEIGHT_res ~ a*h_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_h_PRSres ~~ p_h_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "height_prspc_pa_m1_single.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
HEIGHT_res ~ c*m_h_PRSres
HEIGHT_res ~ b*p_h_PRSres       

#mediator
h_PRSres ~ 0.5*m_h_PRSres
h_PRSres ~ 0.5*p_h_PRSres 

HEIGHT_res ~ a*h_PRSres

p_HEIGHT_res ~ d*p_h_PRSres
m_HEIGHT_res ~ e*m_h_PRSres

HEIGHT_res ~ f*p_HEIGHT_res
HEIGHT_res ~ g*m_HEIGHT_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_h_PRSres ~~ p_h_PRSres
m_HEIGHT_res ~~ p_HEIGHT_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "height_prspc_pa_m2_single.csv", row.names=F, quote=F)

```

Maternal Duo:  

```{r, eval=FALSE}

rm(list=ls())
library(lavaan)

df <- read.csv("HEIGHT_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
HEIGHT_res  ~ c*m_h_PRSres 

# mediator
h_PRSres  ~ 0.5*m_h_PRSres 
HEIGHT_res ~ a*h_PRSres 

# indirect effects (i*a)
ia := 0.5*a

# total effects 
total_mum := c + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "height_prspc_pa_m1_matduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
HEIGHT_res  ~ c*m_h_PRSres

#mediator
h_PRSres ~ 0.5*m_h_PRSres
HEIGHT_res ~ a*h_PRSres
m_HEIGHT_res ~ e*m_h_PRSres
HEIGHT_res ~ g*m_HEIGHT_res

# indirect effects (i*a, e*g)
ia := 0.5*a
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 

# covariance

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "height_prspc_pa_m2_matduo.csv", row.names=F, quote=F)

```

Paternal Duo:  

```{r, eval=FALSE}
rm(list=ls())
library(lavaan)

df <- read.csv("HEIGHT_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
HEIGHT_res ~ b*p_h_PRSres             

# mediator
h_PRSres ~ 0.5*p_h_PRSres
HEIGHT_res ~ a*h_PRSres

# indirect effects (h*a)
ha := 0.5*a

# total effects
total_dad := b + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "height_prspc_pa_m1_patduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
HEIGHT_res ~ b*p_h_PRSres        

#mediator
h_PRSres ~ 0.5*p_h_PRSres 
HEIGHT_res ~ a*h_PRSres
p_HEIGHT_res ~ d*p_h_PRSres
HEIGHT_res ~ f*p_HEIGHT_res

# indirect effects (h*a, d*f)
ha := 0.5*a
df := d*f


# total effects 
total_dad := b + (0.5*a) + (d*f) 

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "height_prspc_pa_m2_patduo.csv", row.names=F, quote=F)
```

## Educational Attainment  

```{r, eval=FALSE}

###############################################################
#             Path h & i restricted to 0.5:  
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("EA_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres
           
# mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres
EA_res ~ a*ea_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_ea_PRSres ~~ p_ea_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "EA_prspc_pa_m1.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres       

#mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres 

EA_res ~ a*ea_PRSres

p_EA_res ~ d*p_ea_PRSres
m_EA_res ~ e*m_ea_PRSres

EA_res ~ f*p_EA_res
EA_res ~ g*m_EA_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_ea_PRSres ~~ p_ea_PRSres
m_EA_res ~~ p_EA_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "EA_prspc_pa_m2.csv", row.names=F, quote=F)


###############################################################
#             Paths freely estimated 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("EA_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres
           
# mediator
ea_PRSres ~ i*m_ea_PRSres
ea_PRSres ~ h*p_ea_PRSres
EA_res ~ a*ea_PRSres

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a

# total effects 
total_mum := c + (i*a)
total_dad := b + (h*a)

# covariance
m_ea_PRSres ~~ p_ea_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "EA_prspc_pa_m1_fm.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres       

#mediator
ea_PRSres ~ i*m_ea_PRSres
ea_PRSres ~ h*p_ea_PRSres 

EA_res ~ a*ea_PRSres

p_EA_res ~ d*p_ea_PRSres
m_EA_res ~ e*m_ea_PRSres

EA_res ~ f*p_EA_res
EA_res ~ g*m_EA_res

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (i*a) + (e*g) 
total_dad := b + (h*a) + (d*f) 

# covariance
m_ea_PRSres ~~ p_ea_PRSres
m_EA_res ~~ p_EA_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "EA_prspc_pa_m2_fm.csv", row.names=F, quote=F)


###############################################################
#             female only results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("EA_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==2)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres
           
# mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres
EA_res ~ a*ea_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_ea_PRSres ~~ p_ea_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "EA_prspc_pa_m1_F.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres       

#mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres 

EA_res ~ a*ea_PRSres

p_EA_res ~ d*p_ea_PRSres
m_EA_res ~ e*m_ea_PRSres

EA_res ~ f*p_EA_res
EA_res ~ g*m_EA_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_ea_PRSres ~~ p_ea_PRSres
m_EA_res ~~ p_EA_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "EA_prspc_pa_m2_F.csv", row.names=F, quote=F)


###############################################################
#             male only results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("EA_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==1)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres
           
# mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres
EA_res ~ a*ea_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_ea_PRSres ~~ p_ea_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "EA_prspc_pa_m1_M.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres       

#mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres 

EA_res ~ a*ea_PRSres

p_EA_res ~ d*p_ea_PRSres
m_EA_res ~ e*m_ea_PRSres

EA_res ~ f*p_EA_res
EA_res ~ g*m_EA_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_ea_PRSres ~~ p_ea_PRSres
m_EA_res ~~ p_EA_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "EA_prspc_pa_m2_M.csv", row.names=F, quote=F)


###############################################################
#             singleton results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("EA_pathwaymodel_df.csv", header=T)
df <- df[!duplicated(df$sibID),]

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres
           
# mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres
EA_res ~ a*ea_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_ea_PRSres ~~ p_ea_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "EA_prspc_pa_m1_single.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
EA_res ~ c*m_ea_PRSres
EA_res ~ b*p_ea_PRSres       

#mediator
ea_PRSres ~ 0.5*m_ea_PRSres
ea_PRSres ~ 0.5*p_ea_PRSres 

EA_res ~ a*ea_PRSres

p_EA_res ~ d*p_ea_PRSres
m_EA_res ~ e*m_ea_PRSres

EA_res ~ f*p_EA_res
EA_res ~ g*m_EA_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_ea_PRSres ~~ p_ea_PRSres
m_EA_res ~~ p_EA_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "EA_prspc_pa_m2_single.csv", row.names=F, quote=F)


###############################################################
#             maternal duo 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("EA_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
EA_res  ~ c*m_ea_PRSres 

# mediator
ea_PRSres  ~ 0.5*m_ea_PRSres 
EA_res ~ a*ea_PRSres 

# indirect effects (i*a)
ab := 0.5*a

# total effects 
total_mum := c + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "EA_prspc_pa_m1_matduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
EA_res  ~ c*m_ea_PRSres

#mediator
ea_PRSres ~ 0.5*m_ea_PRSres
EA_res ~ a*ea_PRSres
m_EA_res ~ e*m_ea_PRSres
EA_res ~ g*m_EA_res

# indirect effects (i*a, e*g)
ia := 0.5*a
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 

# covariance

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "EA_prspc_pa_m2_matduo.csv", row.names=F, quote=F)


###############################################################
#             paternal duo 
###############################################################


rm(list=ls())
library(lavaan)

df <- read.csv("EA_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
EA_res ~ b*p_ea_PRSres             

# mediator
ea_PRSres ~ 0.5*p_ea_PRSres
EA_res ~ a*ea_PRSres

# indirect effects (h*a)
ha := 0.5*a

# total effects
total_dad := b + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "EA_prspc_pa_m1_patduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
EA_res ~ b*p_ea_PRSres        

#mediator
ea_PRSres ~ 0.5*p_ea_PRSres 
EA_res ~ a*ea_PRSres
p_EA_res ~ d*p_ea_PRSres
EA_res ~ f*p_EA_res

# indirect effects (h*a, d*f)
ha := 0.5*a
df := d*f


# total effects 
total_dad := b + (0.5*a) + (d*f) 

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "EA_prspc_pa_m2_patduo.csv", row.names=F, quote=F)

```

## Psychological Distress  

```{r, eval=FALSE}
###############################################################
#             Path h & i restricted to 0.5:  
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQTotal_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQTotal_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQTotal_prspc_pa_m1.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQTotal_res ~ a*dep_PRSres

p_GHQTotal_res ~ d*p_dep_PRSres
m_GHQTotal_res ~ e*m_dep_PRSres

GHQTotal_res ~ f*p_GHQTotal_res
GHQTotal_res ~ g*m_GHQTotal_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQTotal_res ~~ p_GHQTotal_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQTotal_prspc_pa_m2.csv", row.names=F, quote=F)


###############################################################
#             Paths freely estimated 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQTotal_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ i*m_dep_PRSres
dep_PRSres ~ h*p_dep_PRSres
GHQTotal_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a

# total effects 
total_mum := c + (i*a)
total_dad := b + (h*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQTotal_prspc_pa_m1_fm.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ i*m_dep_PRSres
dep_PRSres ~ h*p_dep_PRSres 

GHQTotal_res ~ a*dep_PRSres

p_GHQTotal_res ~ d*p_dep_PRSres
m_GHQTotal_res ~ e*m_dep_PRSres

GHQTotal_res ~ f*p_GHQTotal_res
GHQTotal_res ~ g*m_GHQTotal_res

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (i*a) + (e*g) 
total_dad := b + (h*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQTotal_res ~~ p_GHQTotal_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQTotal_prspc_pa_m2_fm.csv", row.names=F, quote=F)


###############################################################
#             female only results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQTotal_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==2)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQTotal_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQTotal_prspc_pa_m1_F.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQTotal_res ~ a*dep_PRSres

p_GHQTotal_res ~ d*p_dep_PRSres
m_GHQTotal_res ~ e*m_dep_PRSres

GHQTotal_res ~ f*p_GHQTotal_res
GHQTotal_res ~ g*m_GHQTotal_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQTotal_res ~~ p_GHQTotal_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQTotal_prspc_pa_m2_F.csv", row.names=F, quote=F)


###############################################################
#             male only results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQTotal_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==1)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQTotal_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQTotal_prspc_pa_m1_M.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQTotal_res ~ a*dep_PRSres

p_GHQTotal_res ~ d*p_dep_PRSres
m_GHQTotal_res ~ e*m_dep_PRSres

GHQTotal_res ~ f*p_GHQTotal_res
GHQTotal_res ~ g*m_GHQTotal_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQTotal_res ~~ p_GHQTotal_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQTotal_prspc_pa_m2_M.csv", row.names=F, quote=F)


###############################################################
#             singleton results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQTotal_pathwaymodel_df.csv", header=T)
df <- df[!duplicated(df$sibID),]

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQTotal_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQTotal_prspc_pa_m1_single.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQTotal_res ~ c*m_dep_PRSres
GHQTotal_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQTotal_res ~ a*dep_PRSres

p_GHQTotal_res ~ d*p_dep_PRSres
m_GHQTotal_res ~ e*m_dep_PRSres

GHQTotal_res ~ f*p_GHQTotal_res
GHQTotal_res ~ g*m_GHQTotal_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQTotal_res ~~ p_GHQTotal_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQTotal_prspc_pa_m2_single.csv", row.names=F, quote=F)


###############################################################
#             maternal duo 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQTotal_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQTotal_res  ~ c*m_dep_PRSres 

# mediator
dep_PRSres  ~ 0.5*m_dep_PRSres 
GHQTotal_res ~ a*dep_PRSres 

# indirect effects (i*a)
ab := 0.5*a

# total effects 
total_mum := c + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQTotal_prspc_pa_m1_matduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQTotal_res  ~ c*m_dep_PRSres

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
GHQTotal_res ~ a*dep_PRSres
m_GHQTotal_res ~ e*m_dep_PRSres
GHQTotal_res ~ g*m_GHQTotal_res

# indirect effects (i*a, e*g)
ia := 0.5*a
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 

# covariance

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQTotal_prspc_pa_m2_matduo.csv", row.names=F, quote=F)


###############################################################
#             paternal duo 
###############################################################


rm(list=ls())
library(lavaan)

df <- read.csv("GHQTotal_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQTotal_res ~ b*p_dep_PRSres             

# mediator
dep_PRSres ~ 0.5*p_dep_PRSres
GHQTotal_res ~ a*dep_PRSres

# indirect effects (h*a)
ha := 0.5*a

# total effects
total_dad := b + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQTotal_prspc_pa_m1_patduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQTotal_res ~ b*p_dep_PRSres        

#mediator
dep_PRSres ~ 0.5*p_dep_PRSres 
GHQTotal_res ~ a*dep_PRSres
p_GHQTotal_res ~ d*p_dep_PRSres
GHQTotal_res ~ f*p_GHQTotal_res

# indirect effects (h*a, d*f)
ha := 0.5*a
df := d*f


# total effects 
total_dad := b + (0.5*a) + (d*f) 

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQTotal_prspc_pa_m2_patduo.csv", row.names=F, quote=F)
```

## Depression  

```{r, eval=FALSE}
###############################################################
#             Path h & i restricted to 0.5:  
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQDepression_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQDepression_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQDepression_prspc_pa_m1.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQDepression_res ~ a*dep_PRSres

p_GHQDepression_res ~ d*p_dep_PRSres
m_GHQDepression_res ~ e*m_dep_PRSres

GHQDepression_res ~ f*p_GHQDepression_res
GHQDepression_res ~ g*m_GHQDepression_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQDepression_res ~~ p_GHQDepression_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQDepression_prspc_pa_m2.csv", row.names=F, quote=F)


###############################################################
#             Paths freely estimated 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQDepression_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ i*m_dep_PRSres
dep_PRSres ~ h*p_dep_PRSres
GHQDepression_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a

# total effects 
total_mum := c + (i*a)
total_dad := b + (h*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQDepression_prspc_pa_m1_fm.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ i*m_dep_PRSres
dep_PRSres ~ h*p_dep_PRSres 

GHQDepression_res ~ a*dep_PRSres

p_GHQDepression_res ~ d*p_dep_PRSres
m_GHQDepression_res ~ e*m_dep_PRSres

GHQDepression_res ~ f*p_GHQDepression_res
GHQDepression_res ~ g*m_GHQDepression_res

# indirect effects (i*a, h*a)
ia := i*a
ha := h*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (i*a) + (e*g) 
total_dad := b + (h*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQDepression_res ~~ p_GHQDepression_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQDepression_prspc_pa_m2_fm.csv", row.names=F, quote=F)


###############################################################
#             female only results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQDepression_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==2)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQDepression_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQDepression_prspc_pa_m1_F.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQDepression_res ~ a*dep_PRSres

p_GHQDepression_res ~ d*p_dep_PRSres
m_GHQDepression_res ~ e*m_dep_PRSres

GHQDepression_res ~ f*p_GHQDepression_res
GHQDepression_res ~ g*m_GHQDepression_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQDepression_res ~~ p_GHQDepression_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQDepression_prspc_pa_m2_F.csv", row.names=F, quote=F)


###############################################################
#             male only results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQDepression_pathwaymodel_df.csv", header=T)
df <- subset(df, df$SEX==1)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQDepression_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQDepression_prspc_pa_m1_M.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQDepression_res ~ a*dep_PRSres

p_GHQDepression_res ~ d*p_dep_PRSres
m_GHQDepression_res ~ e*m_dep_PRSres

GHQDepression_res ~ f*p_GHQDepression_res
GHQDepression_res ~ g*m_GHQDepression_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQDepression_res ~~ p_GHQDepression_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQDepression_prspc_pa_m2_M.csv", row.names=F, quote=F)


###############################################################
#             singleton results 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQDepression_pathwaymodel_df.csv", header=T)
df <- df[!duplicated(df$sibID),]

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres
           
# mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres
GHQDepression_res ~ a*dep_PRSres

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a

# total effects 
total_mum := c + (0.5*a)
total_dad := b + (0.5*a)

# covariance
m_dep_PRSres ~~ p_dep_PRSres
'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQDepression_prspc_pa_m1_single.csv", row.names=F, quote=F)


################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQDepression_res ~ c*m_dep_PRSres
GHQDepression_res ~ b*p_dep_PRSres       

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
dep_PRSres ~ 0.5*p_dep_PRSres 

GHQDepression_res ~ a*dep_PRSres

p_GHQDepression_res ~ d*p_dep_PRSres
m_GHQDepression_res ~ e*m_dep_PRSres

GHQDepression_res ~ f*p_GHQDepression_res
GHQDepression_res ~ g*m_GHQDepression_res

# indirect effects (i*a, h*a)
ia := 0.5*a
ha := 0.5*a
df := d*f
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 
total_dad := b + (0.5*a) + (d*f) 

# covariance
m_dep_PRSres ~~ p_dep_PRSres
m_GHQDepression_res ~~ p_GHQDepression_res

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQDepression_prspc_pa_m2_single.csv", row.names=F, quote=F)


###############################################################
#             maternal duo 
###############################################################

rm(list=ls())
library(lavaan)

df <- read.csv("GHQDepression_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQDepression_res  ~ c*m_dep_PRSres 

# mediator
dep_PRSres  ~ 0.5*m_dep_PRSres 
GHQDepression_res ~ a*dep_PRSres 

# indirect effects (i*a)
ab := 0.5*a

# total effects 
total_mum := c + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQDepression_prspc_pa_m1_matduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQDepression_res  ~ c*m_dep_PRSres

#mediator
dep_PRSres ~ 0.5*m_dep_PRSres
GHQDepression_res ~ a*dep_PRSres
m_GHQDepression_res ~ e*m_dep_PRSres
GHQDepression_res ~ g*m_GHQDepression_res

# indirect effects (i*a, e*g)
ia := 0.5*a
eg := e*g


# total effects 
total_mum := c + (0.5*a) + (e*g) 

# covariance

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQDepression_prspc_pa_m2_matduo.csv", row.names=F, quote=F)


###############################################################
#             paternal duo 
###############################################################


rm(list=ls())
library(lavaan)

df <- read.csv("GHQDepression_pathwaymodel_df.csv", header=T)

##############
#SIMPLE MODEL#
##############

# Define model

model <- ' 

# direct effects
GHQDepression_res ~ b*p_dep_PRSres             

# mediator
dep_PRSres ~ 0.5*p_dep_PRSres
GHQDepression_res ~ a*dep_PRSres

# indirect effects (h*a)
ha := 0.5*a

# total effects
total_dad := b + (0.5*a)

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')

sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
simple_model <- cbind(sem_model_final, AIC(fitM))

write.csv(simple_model, "GHQDepression_prspc_pa_m1_patduo.csv", row.names=F, quote=F)

################
#EXTENDED MODEL#
################

# Define model

model <- '

#direct effects
GHQDepression_res ~ b*p_dep_PRSres        

#mediator
dep_PRSres ~ 0.5*p_dep_PRSres 
GHQDepression_res ~ a*dep_PRSres
p_GHQDepression_res ~ d*p_dep_PRSres
GHQDepression_res ~ f*p_GHQDepression_res

# indirect effects (h*a, d*f)
ha := 0.5*a
df := d*f


# total effects 
total_dad := b + (0.5*a) + (d*f) 

'

# Fit the model
fitM <- sem(model, data=df, estimator='ML')
sem_model <- summary(fitM, standardized=TRUE, fit.measures=TRUE, ci=TRUE)
sem_model_df <- as.data.frame(sem_model$PE)
sem_model_final <- sem_model_df[sem_model_df$label != "", c(1:4,6:7,9:11)] #here we are extracting values of interest e.g. est, se, p, ci
extended_model <- cbind(sem_model_final, AIC(fitM))

write.csv(extended_model, "GHQDepression_prspc_pa_m2_patduo.csv", row.names=F, quote=F)
```


